lib/python/picongpu/plugins/plot_mpl/emittance_evolution_visualizer.py

```diff
@@ -0,0 +1,158 @@
+
+    def _create_plt_obj(self, ax):
+        """
+        Implementation of base class function.
+        Turns 'self.plt_obj' into a matplotlib.pyplot.plot object.
+        """
+        emit, y_slices, iteration = self.data
+        np_data = np.zeros(len(iteration))
+        for index, ts in enumerate(iteration):
+            np_data[index] = emit[ts][0]
+        self.plt_obj, = ax.plot(iteration * 1.39e-16 * 1.e12,
+                                np_data * 1.e6, scalex=True, scaley=True)
+        self.plt_lin = ax.axvline(self.itera * 1.39e-16 * 1.e12,
+                                  color='#FF6600')
+
+    def _update_plt_obj(self):
+        """
+        Implementation of base class function.
+        """
+        emit, y_slices, iteration = self.data
+        np_data = np.zeros(len(iteration))
+        for index, ts in enumerate(iteration):
+            np_data[index] = emit[ts][0]
+        if self.plt_lin:
+            self.plt_lin.remove()
+        self.plt_obj.set_data(iteration * 1.39e-16 * 1.e12, np_data*1.e6)
+        self.plt_lin = self.ax.axvline(self.itera * 1.39e-16 * 1.e12,
+                                       color='#FF6600')
+        self.ax.relim()
+        self.ax.autoscale_view(True, True, True)
+
+    def visualize(self, ax=None, **kwargs):
+        """
+        Creates a semilogy plot on the provided axes object for
+        the data of the given iteration using matpotlib.
+
+        Parameters
+        ----------
+        iteration: int
+            the iteration number for which data will be plotted.
+        ax: matplotlib axes object
+            the part of the figure where this plot will be shown.
+        kwargs: dictionary with further keyword arguments, valid are:
+            species: string
+                short name of the particle species, e.g. 'e' for electrons
+                (defined in ``speciesDefinition.param``)
+            iteration: int
+                number of the iteration
+            species_filter: string
+                name of the particle species filter, default is 'all'
+                (defined in ``particleFilters.param``)
+
+        """
+        self.ax = self._ax_or_gca(ax)
+        self.itera = kwargs.get('iteration')
+        # this already throws error if no species or iteration in kwargs
+        kwargs['iteration'] = None
+        super(Visualizer, self).visualize(ax, **kwargs)
+        species = kwargs.get('species')
+        species_filter = kwargs.get('species_filter', 'all')
+        ax.set_xlabel('time [ps]')
+        ax.set_ylabel(r'emittance [$\mathrm{\pi mm mrad}$]')
+        ax.set_title('emittance for species ' +
+                     species + ', filter = ' + species_filter)
+
+
+if __name__ == '__main__':
+
+    def main():
+        import sys
+        import getopt
+
+        def usage():
+            print("usage:")
+            print(
+                "python", sys.argv[0],
+                "-p <path to run_directory> -i <iteration>"
```

Comment:
I think iteration should be optional for general usage (just do not plot the progress line if not given)